### PR TITLE
SAML Setup

### PR DESCRIPTION
**Pull request by mattkrick**
- tests to be run on localhost:3000, since that's where the redirect is pointing to.
- run setup as shown in https://github.com/ParabolInc/action-devops/issues/37

TEST
- [ ] at login screen, can type email & click login. it redirect to okta
- [ ] after successful login with okta (see private repo for details) it creates a new user
- [ ] can log out
- [ ] can log in again & it works
- [ ] log out & then visit a mass invite link. can log in via that modal, too
- [ ] as a 2nd user, send a team invite to the SSO user, have the url printed to the server console
- [ ] log out & visit the team invite link. It knows you're an SSO user & automatically opens okta for you & attempts to log in
- [ ] delete user in rethinkdb. then, repeat team invite link step. accepting a team invite as a new user gives you the suggested action of trying the demo
 
@jordanh

### CODE REVIEW

File: packages/client/components/GenericAuthentication.tsx [hunk 1]
```diff
@@ -178,7 +178,9 @@ class GenericAuthentication extends Component<Props, State> {
         {isCreate ? (
           <AuthPrivacyFooter />
         ) : (
+          <>
```
Comment (jordanh) [inline packages/client/components/GenericAuthentication.tsx]: What's this empty markup here?

File: packages/client/components/Tag/TierTag.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  backgroundColor: PALETTE.BACKGROUND_MAIN,
+  color: PALETTE.TEXT_MAIN
+})
+
+const ProTag = styled(BaseTag)({
+  backgroundColor: PALETTE.BACKGROUND_YELLOW,
+  color: PALETTE.TEXT_PURPLE
+})
+
+const EnterpriseTag = styled(BaseTag)({
```
Comment (jordanh) [inline packages/client/components/Tag/TierTag.tsx]: We all knew Enterprise was blue

File: packages/server/database/migrations/20170926142800-freemium.js [hunk 3]
```diff
@@ -18,13 +17,13 @@ exports.up = async (r) => {
         r
           .table('Organization')
           .get(org('id'))
-          .replace((row) => row.merge({tier: PERSONAL}).without('stripeId', 'stripeSubscriptionId'))
+          .replace((row) => row.merge({tier: 'personal'}).without('stripeId', 'stripeSubscriptionId'))
```
Comment (jordanh) [inline packages/server/database/migrations/20170926142800-freemium.js]: Any reason these went from vars to hardcoded strings?

File: packages/server/database/migrations/20180509162156-addFreeRetros.js [hunk 4]
```diff
@@ -1,10 +1,8 @@
-import {PRO} from '../../../client/utils/constants'
-
 exports.up = async (r) => {
   try {
     await r.table('Organization').update((org) => {
       return r.branch(
-        org('tier').eq(PRO),
+        org('tier').eq('pro'),
```
Comment (jordanh) [inline packages/server/database/migrations/20180509162156-addFreeRetros.js]: Ditto this

File: packages/server/database/migrations/20190904182024-saml.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+exports.up = async (r) => {
+  try {
+    await r.tableCreate('SAML')
```
Comment (jordanh) [inline packages/server/database/migrations/20190904182024-saml.js]: Ah interesting, you decided to put it on its own table. Wasn't sure if it should go here, or on an org. Given 1 IdP can have many orgs, this makes sense

File: packages/server/utils/consumeSAML.ts [hunk 6]
```diff
@@ -0,0 +1,85 @@
+import getRethink from '../database/rethinkDriver'
```
Comment (jordanh) [inline packages/server/utils/consumeSAML.ts]: Ah, so this is where the magic happens

**Comment by jordanh** [comment]
# Test Cycle 1

## Tests
- [x] at login screen, can type email & click login. it redirect to okta
- [ ] after successful login with okta (see private repo for details) it creates a new user
- [ ] can log out
- [ ] can log in again & it works
- [ ] log out & then visit a mass invite link. can log in via that modal, too
- [ ] as a 2nd user, send a team invite to the SSO user, have the url printed to the server console
- [ ] log out & visit the team invite link. It knows you're an SSO user & automatically opens okta for you & attempts to log in
- [ ] delete user in rethinkdb. then, repeat team invite link step. accepting a team invite as a new user gives you the suggested action of trying the demo

File: packages/server/database/migrations/20170926142800-freemium.js (hunk 3 above)
Comment (mattkrick) [reply packages/server/database/migrations/20170926142800-freemium.js]: got rid of the old constants so had to nuke everything that had a ref to them

File: packages/client/components/GenericAuthentication.tsx (hunk 1 above)
Comment (mattkrick) [reply packages/client/components/GenericAuthentication.tsx]: shorthand for `<React.Fragment>`

**Comment by jordanh** [comment]
gif

**Pull request merged**